File: libraries/flutter_inapp_purchase/lib/deprecation.dart
import 'package:flutter/foundation.dart';

final Set<String> _emittedLegacyWarnings = <String>{};

/// Emits a compatibility warning once per legacy wire shape.
///
/// This intentionally remains separate from debug logging because callers
/// need to see migration guidance before the compatibility path is removed.
void warnLegacyOnce(String key, String message) {
  if (!_emittedLegacyWarnings.add(key)) {
    return;
  }

  debugPrint('[flutter_inapp_purchase] DEPRECATED: $message');
}

@visibleForTesting
void resetLegacyWarningsForTesting() {
  _emittedLegacyWarnings.clear();
}

File: libraries/expo-iap/src/utils/deprecation.ts
import {ExpoIapConsole} from './debug';

const emittedLegacyWarnings = new Set<string>();

export const warnLegacyOnce = (key: string, message: string): void => {
  if (emittedLegacyWarnings.has(key)) {
    return;
  }

  emittedLegacyWarnings.add(key);
  ExpoIapConsole.warn(message);
};

export const resetLegacyWarningsForTesting = (): void => {
  emittedLegacyWarnings.clear();
};
